packages/client/lib/client/pool.ts

```diff
@@ -288,31 +310,43 @@ export class RedisClientPool<
 
   async connect() {
     if (this._self.#isOpen) return; // TODO: throw error?
-
     this._self.#isOpen = true;
 
+    try {
+      this._self.#clientSideCache?.onPoolConnect(this._self.#clientFactory);
+    } catch (err) {
+      this.destroy();
+      throw err;
+    }
+
```

Comment:
this was only to support redirect situation, which isn't happening, so should be removed to simplify code.  and remove onPoolConnect() from interface.